mne/decoding/time_gen.py

```diff
@@ -914,3 +953,358 @@ def time_generalization(epochs_list, clf=None, cv=5, scoring="roc_auc",
+            Defaults to None.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        from sklearn.base import clone
+        from sklearn.cross_validation import check_cv, StratifiedKFold
+
+        n_jobs = self.n_jobs
+
+        if picks is None:
+            picks = pick_types(epochs.info, meg=True, eeg=True,
+                               exclude='bads')
+
+        event_ids = epochs.event_id
+        epochs_list = [epochs[event] for event in event_ids]
+        equalize_epoch_counts(epochs_list)
+
+        self.times_ = epochs.times
+        X = [e.get_data()[:, picks, :] for e in epochs_list]
+        if y is None:
+            y = np.concatenate(
+                [k * np.ones(len(this_X)) for k, this_X in enumerate(X)])
+        self.y_train_ = y
+        X = np.concatenate(X)
+
+        # Chunk X for parallelization
+        if n_jobs > 0:
+            n_chunk = n_jobs
+        else:
+            n_chunk = multiprocessing.cpu_count()
+
+        # Avoid splitting the data in more time chunk than there is time points
+        if n_chunk > X.shape[2]:
+            n_chunk = X.shape[2]
+
+        cv = self.cv
+        if isinstance(cv, (int, np.int)):
+            cv = StratifiedKFold(y, cv)
+        cv = check_cv(cv, X, y, classifier=True)
+        self.cv_ = cv  # update CV
+
+        def _cv_constant_time(clf, X, y, cv):
+            estimators = list()
+            for train, test in cv:
+                clf.fit(X[train, :], y[train])
+                estimators.append(clf)
+            return estimators
+
+        parallel, parallel_time, n_jobs = parallel_func(
+            _cv_constant_time, n_jobs)
+
+        # Parallelize across n_times.
+        n_times = epochs.times.size
+        self.estimators_ = parallel(
+            parallel_time(clone(self.clf), X[:, :, i], y, cv)
+            for i in xrange(n_times))
+        return self
+
+    def predict(self, epochs, picks=None, test_times=None):
+        """ Predict the event_id for every epoch for every time point.
+
+        Note. This function sets and updates the ``y_pred_`` attribute.
+
+        Parameters
+        ----------
+        epochs : instance of Epochs
+            The epochs. Can be similar to fitted epochs or not. See independent
+            parameter.
+
+        picks : np.ndarray (n_selected_chans,) | None
+            Channels to be included.
+
+        test_times : None
+            Not to be used.
+
+        Returns
+        -------
+        y_pred_ : np.ndarray, shape (n_times, n_epochs, n_prediction_dims)
+            Class labels for samples in X.
+        """
+        from scipy import stats
+        n_jobs = self.n_jobs
+
+        if not hasattr(self, 'estimators_'):
+            raise ValueError("Estimators have to be fit before predicting.")
+
+        if picks is None:
+            picks = pick_types(epochs.info, meg=True, eeg=True,
```

Comment:
we should probably have some sort of check to see if the `info` object is the same for training and testing